### ISSUE TITLE
Problems generating BakedLightMaps by code

### ISSUE THREAD

**Issue by maikramer** [godotengine/godot#18954]
**Godot version:**
ba134d44b8a5d87cf3694dd8858ec2a495bf6c11

**OS/device including version:**
Ubuntu 18.04

**Issue description:**
While trying to call the function BakedLightMap,Bake() i get in the console
``` 
ERROR: write_variant: Resource was not pre cached for the resource section, bug?
At: core/io/resource_format_binary.cpp:1508.
```
I debugged athe code and detected that the problem happens in 
https://github.com/godotengine/godot/blob/ba134d44b8a5d87cf3694dd8858ec2a495bf6c11/core/io/resource_saver.cpp#L74

This happens while BakedLightMap tryes to save the ImageTexture in disk.

I guess that this issue is related to #18801, but as they are 2 totally diferent situations i decided to open a new issue.

**Steps to reproduce:**
Bake a lightmap by code.

**Comment by reduz** [comment]
This needs to be re-made in 3.2 with Vulkan, will kick there

**Comment by KoBeWi** [comment]
Seems fixed in 3.2.4 beta3 (no error on calling `bake()`).
Can't test in 4.0, because `bake()` method doesn't exist for some reason (even though you can see it in online docs)

If you can still reproduce, provide a minimal reproduction project.
